python/cuml/preprocessing/encoders.py

```diff
@@ -281,44 +290,77 @@ def transform(self, X):
         X = self._check_input(X)
 
         cols, rows = list(), list()
+        col_idx = None
         j = 0
-        for feature in X.columns:
-            encoder = self._encoders[feature]
-            col_idx = encoder.transform(X[feature])
-            idx_to_keep = cp.asarray(col_idx.notnull().to_gpu_array())
-            col_idx = cp.asarray(col_idx.dropna().to_gpu_array())
-
-            # increase indices to take previous features into account
-            col_idx += j
-
-            # Filter out rows with null values
-            row_idx = cp.arange(len(X))[idx_to_keep]
-
-            if self.drop_idx_ is not None:
-                drop_idx = self.drop_idx_[feature] + j
-                mask = cp.ones(col_idx.shape, dtype=cp.bool)
-                mask[col_idx == drop_idx] = False
-                col_idx = col_idx[mask]
-                row_idx = row_idx[mask]
-                # account for dropped category in indices
-                col_idx[col_idx > drop_idx] -= 1
-                # account for dropped category in current cats number
-                j -= 1
-            j += len(encoder.classes_)
-            cols.append(col_idx)
-            rows.append(row_idx)
-
-        cols = cp.concatenate(cols)
-        rows = cp.concatenate(rows)
-        val = cp.ones(rows.shape[0], dtype=self.dtype)
-        ohe = cp.sparse.coo_matrix((val, (rows, cols)),
-                                   shape=(len(X), j),
-                                   dtype=self.dtype)
 
-        if not self.sparse:
-            ohe = ohe.toarray()
+        try:
+            for feature in X.columns:
+                encoder = self._encoders[feature]
+                col_idx = encoder.transform(X[feature])
+                idx_to_keep = cp.asarray(col_idx.notnull().to_gpu_array())
+                col_idx = cp.asarray(col_idx.dropna().to_gpu_array())
+
+                # Force uint64 dtype if specified
+                if (self.force_uint64_cols):
```

Comment:
I think you can get rid of this option due to the dynamic upscaling

Reply:
Done